codegen/src/test/resources/software/amazon/awssdk/codegen/poet/auth/scheme/s3-test-auth-scheme-default-provider.java

```diff
@@ -0,0 +1,142 @@
+                                                              .useS3ExpressControlEndpoint(params.useS3ExpressControlEndpoint())
+                                                              .disableS3ExpressSessionAuth(params.disableS3ExpressSessionAuth()).deleteObjectKeys(params.deleteObjectKeys())
+                                                              .build();
+        Endpoint endpoint = CompletableFutureUtils.joinLikeSync(endpointProvider(params).resolveEndpoint(endpointParameters));
+        List<EndpointAuthScheme> authSchemes = endpoint.attribute(AwsEndpointAttribute.AUTH_SCHEMES);
+        if (authSchemes == null) {
+            return FALLBACK_RESOLVER.resolveAuthScheme(params);
+        }
+        List<AuthSchemeOption> options = new ArrayList<>();
+        for (EndpointAuthScheme authScheme : authSchemes) {
+            String name = authScheme.name();
+            switch (name) {
+                case "sigv4":
+                    SigV4AuthScheme sigv4AuthScheme = Validate.isInstanceOf(SigV4AuthScheme.class, authScheme,
+                                                                            "Expecting auth scheme of class SigV4AuthScheme, got instead object of class %s", authScheme.getClass()
+                                                                                                                                                                        .getName());
+                    AuthSchemeOption sigv4AuthSchemeOption = applySigV4FamilyDefaults(
+                        AuthSchemeOption.builder().schemeId(AwsV4AuthScheme.SCHEME_ID)
+                                        .putSignerProperty(AwsV4HttpSigner.SERVICE_SIGNING_NAME, sigv4AuthScheme.signingName())
+                                        .putSignerProperty(AwsV4HttpSigner.REGION_NAME, sigv4AuthScheme.signingRegion())
+                                        .putSignerProperty(AwsV4HttpSigner.DOUBLE_URL_ENCODE, !sigv4AuthScheme.disableDoubleEncoding()),
+                        params).build();
+                    options.add(sigv4AuthSchemeOption);
+                    break;
+                case "sigv4a":
+                    SigV4aAuthScheme sigv4aAuthScheme = Validate.isInstanceOf(SigV4aAuthScheme.class, authScheme,
+                                                                              "Expecting auth scheme of class SigV4AuthScheme, got instead object of class %s", authScheme.getClass()
+                                                                                                                                                                          .getName());
+                    RegionSet regionSet = Optional.ofNullable(params.regionSet()).orElseGet(
+                        () -> Optional.ofNullable(sigv4aAuthScheme.signingRegionSet())
+                                      .filter(set -> !CollectionUtils.isNullOrEmpty(set)).map(RegionSet::create).orElse(null));
```

Comment:
Unrelated to this change, but, let's say, this can be improved, two allocs just to write this in a single expression is not a good tradeoff, in my not-so-humble opinion.